From ffd2df3cc027d9416c974f410ee62d0b71f3500b Mon Sep 17 00:00:00 2001
From: Alexander Martinz <amartinz@shiftphones.com>
Date: Wed, 27 Feb 2019 16:01:53 +0100
Subject: [PATCH 19/20] no_results: also tint image drawable

If we are setting color for it, the drawable got ignored, resulting
in eg white text and a black drawable.

Tint drawable as well when setting text to make all match.

Change-Id: I29cda801a0319699bb87213bc702e3a28ec4fae8
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 res/layout/no_results_message.xml                        | 1 +
 src/org/lineageos/eleven/widgets/NoResultsContainer.java | 7 +++++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/res/layout/no_results_message.xml b/res/layout/no_results_message.xml
index cf78c5b..9d32ba6 100644
--- a/res/layout/no_results_message.xml
+++ b/res/layout/no_results_message.xml
@@ -27,6 +27,7 @@
     android:visibility="gone">
 
     <ImageView
+        android:id="@+id/no_results_image"
         android:layout_width="@dimen/white_note_width"
         android:layout_height="@dimen/white_note_height"
         android:layout_gravity="center_horizontal"
diff --git a/src/org/lineageos/eleven/widgets/NoResultsContainer.java b/src/org/lineageos/eleven/widgets/NoResultsContainer.java
index 26b569b..770afc3 100644
--- a/src/org/lineageos/eleven/widgets/NoResultsContainer.java
+++ b/src/org/lineageos/eleven/widgets/NoResultsContainer.java
@@ -16,8 +16,10 @@
 package org.lineageos.eleven.widgets;
 
 import android.content.Context;
+import android.graphics.PorterDuff;
 import android.util.AttributeSet;
 import android.view.View;
+import android.widget.ImageView;
 import android.widget.LinearLayout;
 import android.widget.TextView;
 
@@ -43,7 +45,7 @@ public class NoResultsContainer extends LinearLayout {
     }
 
     public void setMainHighlightText(final String text) {
-        final TextView hightlightText = (TextView)findViewById(R.id.no_results_main_highlight_text);
+        final TextView hightlightText = findViewById(R.id.no_results_main_highlight_text);
 
         if (text == null || text.isEmpty()) {
             hightlightText.setVisibility(View.GONE);
@@ -58,8 +60,9 @@ public class NoResultsContainer extends LinearLayout {
     }
 
     public void setTextColor(int color) {
+        ((ImageView)findViewById(R.id.no_results_image)).setColorFilter(color, PorterDuff.Mode.SRC_IN);
         ((TextView)findViewById(R.id.no_results_main_text)).setTextColor(color);
         ((TextView)findViewById(R.id.no_results_main_highlight_text)).setTextColor(color);
         ((TextView)findViewById(R.id.no_results_secondary_text)).setTextColor(color);
     }
-}
\ No newline at end of file
+}
-- 
2.17.1

